File: NEsper.IO/OutputAdapter.cs
using System;
using System.Collections.Generic;

using net.esper.adapter.subscription;

namespace net.esper.adapter
{
	/// <summary>
	/// An output adapter transforms engine events and
	/// </summary>
	public interface OutputAdapter : Adapter
	{
	    /// <summary>
	    /// Sets the subscriptions for the output adapter.
	    /// </summary>
	    /// <param name="subscriptionMap">is the active subscriptions.</param>
	    void setSubscriptionMap(IDictionary<String, Subscription> subscriptionMap);

	    /// <summary>
	    /// Returns the subscriptions.
	    /// <returns>map of name and subscription</returns>
	    /// </summary>
	    IDictionary<String, Subscription> getSubscriptionMap();

	    /// <summary>
	    /// Returns a given subscription by it's name, or null if not found
	    /// </summary>
	    /// <param name="subscriptionAlias">is the subscription</param>
	    /// <returns>subcription or null</returns>
	    Subscription getSubscription(String subscriptionAlias);
	}
}

File: trunk/esperio/src/main/java/com/espertech/esper/adapter/OutputAdapter.java
package com.espertech.esper.adapter;

import com.espertech.esper.adapter.subscription.*;

import java.util.*;

/**
 * An output adapter transforms engine events and
 */
public interface OutputAdapter extends Adapter
{
    /**
     * Sets the subscriptions for the output adapter.
     * @param subscriptionMap is the active subscriptions.
     */
    public void setSubscriptionMap(Map<String, Subscription> subscriptionMap);

    /**
     * Returns the subscriptions.
     * @return map of name and subscription
     */
    public Map<String, Subscription> getSubscriptionMap();

    /**
     * Returns a given subscription by it's name, or null if not found
     * @param subscriptionAlias is the subscription
     * @return subcription or null
     */
    public Subscription getSubscription(String subscriptionAlias);
}
